frontend/packages/pipelines-plugin/src/components/shared/common/auto-complete/autoCompleteValueParsers.ts

```diff
@@ -0,0 +1,48 @@
+import { useFormikContext } from 'formik';
+import { PipelineTask } from '../../../../types';
+import {
+  PipelineBuilderFormikValues,
+  SelectedBuilderTask,
+} from '../../../pipelines/pipeline-builder/types';
+import {
+  findTasksThatRunAfter,
+  paramToAutoComplete,
+  taskToResult,
+  taskToStatus,
+  workspaceToAutoComplete,
+} from './autoCompleteUtils';
+
+export const useBuilderParams = (selectedData: SelectedBuilderTask): string[] => {
```

Comment:
You'll want to make one of these for the WhenExpressions auto complete values @karthikjeeyar -- not sure you need much more than that right now. We can connect it to the AutoCompletePopover once the kinks are worked out here.